### PR TITLE
update @expo/vector-icons and react-native-vector-icons types to add new static methods and AntDesign icons

### PR DESCRIPTION
**Pull request by jkillian**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/oblador/react-native-vector-icons/releases/tag/v5.0.0, https://github.com/oblador/react-native-vector-icons/releases/tag/v6.0.0, https://github.com/expo/vector-icons/commit/bf49299526da3d902cb89579e9ba1798ec00b7ef
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native-vector-icons/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "dom"
+            "es6"
```
Comment (JKillian) [inline types/react-native-vector-icons/tsconfig.json]: had to change this to get the lint to pass, not sure why it was set to "dom"

**Review by iRoachie — APPROVED** [review]

**Review by incleaf — APPROVED** [review]

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/expo__vector-icons/index.d.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
-// Type definitions for @expo/vector-icons 6.2
+// Type definitions for @expo/vector-icons 8.0
```
[the remark this answers is not in the document]
Comment (JKillian) [reply types/expo__vector-icons/index.d.ts]: Happy to follow those steps. Just one question @rbuckton, the docs say:

> If you intend to continue updating the older version of the package, you may create a new subfolder with the current version 

In this case, I doubt maintaining older versions of types matters, it it necessary to create a subfolder in that case?

**Comment by JKillian** [comment]
I'm pretty sure [this code](https://github.com/Microsoft/types-publisher/blob/10294c0e763cc63a5503f5782f31563ed7f80e71/src/lib/definition-parser.ts#L286-L291) makes it impossible to do path mappings correctly for scoped and versioned packages. Going to investigate further though

File: types/expo__vector-icons/index.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-ms) [reply types/expo__vector-icons/index.d.ts]: No, you shouldn't create it if you won't use it.

**Pull request closed without merging**